### PR TITLE
fix(ivy): DebugNode throws exceptions when querying some properties

### PR DESCRIPTION
**Pull request by atscott**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

In Ivy, DebugElement.query throws exceptions when querying some properties because it encounters elements created outside Angular's context.

Issue Number: N/A

## What is the new behavior?

No exceptions are thrown

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
DebugElement.injector still throws an exception because the interface doesn't support returning null.

### CODE REVIEW

File: packages/core/src/debug/debug_node.ts [hunk 1]
```diff
@@ -217,24 +219,39 @@ class DebugNode__POST_R3__ implements DebugNode {
...
+    return this.hasPatchedData(this.nativeNode) ? getLocalRefs(this.nativeNode) : {};
+  }
+
+  get providerTokens(): any[] {
+    return this.hasPatchedData(this.nativeNode) ? getInjectionTokens(this.nativeNode as Element) :
+                                                  [];
+  }
 
-  get providerTokens(): any[] { return getInjectionTokens(this.nativeNode as Element); }
+  private hasPatchedData(target: any) { return target && readPatchedData(target) !== null; }
```
Comment (AndrewKushnir) [inline packages/core/src/debug/debug_node.ts]: Small nit: we can probably move this function outside of the calls (as it doesn't use `this` inside) and add a return value type:
```
function hasPatchedData(target: any): boolean {
  return !!target && readPatchedData(target) !== null;
}
```
Another option might be to co-locate this function in the `render3/util/view_utils` file alongside the `readPatchedData` function and just check if property exists:
```
function hasPatchedData(target: any): boolean {
  return !!target && target.hasOwnProperty(MONKEY_PATCH_KEY_NAME);
}
```

Comment (atscott) [reply packages/core/src/debug/debug_node.ts]: Done.

Comment (AndrewKushnir) [reply packages/core/src/debug/debug_node.ts]: Thanks for the update 👍

**Review by kara — CHANGES_REQUESTED** [review]

**Review by kara — APPROVED** [review]
LGTM

**Pull request closed without merging**